src/elements/content-sidebar/ActivitySidebar.js

```diff
@@ -191,9 +191,17 @@ class ActivitySidebar extends React.PureComponent<Props, State> {
             // need to load the pending item
             this.fetchFeedItems();
         },
-        updateTask: ({ text, id }: { id: string, text: string }): void => {
-            const { file, api } = this.props;
-            api.getFeedAPI(false).updateTask(file, id, text, this.feedSuccessCallback, this.feedErrorCallback);
+        updateTask: (task: TaskUpdatePayload): void => {
+            const { file, api, onTaskUpdate = noop } = this.props;
+            api.getFeedAPI(false).updateTaskNew(
+                file,
+                task,
+                () => {
+                    this.feedSuccessCallback();
```

Comment:
Should this logic be moved to its own named method so it can be tested? Looks like it's missing in the unit tests right now.

Reply:
I would like this be inline for now, this will be refactored in the next chunk of work.